### PR TITLE
[Sema] Don't early exit without looking at conformances in ApplyClassifier

### LINKED ISSUE #60351: [main] Regression when awaiting `async let`s that are re-throwing
**Describe the bug**
The following code compiles perfectly fine on 5.6 and 5.7; however, it is broken on recent `main`.

```swift
func foo() async {
    let stream = AsyncStream<Int>{ _ in }
    async let bar = stream.first { _ in true}

    await bar // error: call can throw, but it is not marked with 'try' and the error is not handled
}
```

**Expected behavior**
The above code should compile on main and should not require a `try` since the call to `first` is re throwing.

rdar://97966043

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
Previous change to apply classifier caused a regression where previously valid code 
```swift
func search(query: String, entities: [String]) async throws -> [String] {
  async let r = entities.filter { $0.contains(query) }.map { String($0) }
  return await r 
}
```
was now being rejected. 

This reverts the change for now, but we the proper fix should be adjust apply classifier to handle this correctly.

Edit: Conformances were not being taking into account because of early return, so this now continues if there are any conformances that need to be looked at. 

Resolves: https://github.com/apple/swift/issues/60351

### CODE REVIEW

File: test/Concurrency/sr15049.swift [hunk 1]
```diff
@@ -49,3 +51,8 @@ func testAsyncSequence3<Seq>(_ seq: Seq) async throws where Seq: AsyncSequence,
 func testAsyncSequence4<Seq>(_ seq: Seq) async throws where Seq: AsyncSequence, Seq.Element == Int { // expected-note{{consider making generic parameter 'Seq' conform to the 'Sendable' protocol}} {{28-28=: Sendable}}
    async let _ = seq // expected-warning{{capture of 'seq' with non-sendable type 'Seq' in 'async let' binding}}
 }
+
+func search(query: String, entities: [String]) async throws -> [String] {
+  async let r = entities.filter { $0.contains(query) }.map { String($0) }
+  return await r // OK
```
Comment (xedin) [inline test/Concurrency/sr15049.swift:57]: Thank you for including the test-case!

**Comment by LucianoPAlmeida** [comment]
@DougGregor a friendly ping for a review =]

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please test

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please test

**Comment by LucianoPAlmeida** [comment]
@swift-ci Please test

**Review by ktoso — APPROVED** [review]
Looks good to me :)

**Comment by ktoso** [comment]
Resolved https://github.com/apple/swift/issues/60351

**Pull request merged**